### PR TITLE
feat: update PR target automatically when parent PR is integrated

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- When the parent branch is merged and the user hits "update", we can handle updating the target branch of the PR for the user.
- Previously, they had to have the GitHub repository setting "delete branch once merged" enabled. This change means the user will no longer have to have this  enabled.

### CODE REVIEW

File: apps/desktop/src/lib/branch/SeriesHeader.svelte [hunk 1]
```diff
@@ -165,6 +166,28 @@
 		}
 	});
 
+	/**
+	 * If the repository does not have "delete after merged" enabled, we need to manually update the base
+	 * of the bottom most branch to point to the projects baseBranch of choice after its parent has been integrated,
+	 * and  it now is the bottom-most series.
+	 */
+	$effect(() => {
```
Comment (ndom91) [inline apps/desktop/src/lib/branch/SeriesHeader.svelte:177]: Not super happy with putting this here, open to any other suggestions

**Comment by krlvi** [comment]
Nice one!

> Previously, they had to have the GitHub repository setting "delete branch once merged" enabled. This change means the user will no longer have to have this enabled.

As long as merging didnt happen on github itself :p

File: apps/desktop/src/lib/branch/SeriesHeader.svelte [hunk 2]
```diff
@@ -163,6 +164,28 @@
...
+		const targetBase = $baseBranch.branchName.replace(`${$baseBranch.remoteName}/`, '');
+		if (
+			listedPr?.targetBranch !== targetBase &&
+			isBottomSeries &&
```
Comment (mtsgrd) [inline apps/desktop/src/lib/branch/SeriesHeader.svelte:181]: Let's name this  `isBottomBranch`, rename and `isTopSeries` to `isTopBranch`?

File: apps/desktop/src/lib/branch/SeriesHeader.svelte (hunk 1 above)
Comment (mtsgrd) [reply apps/desktop/src/lib/branch/SeriesHeader.svelte:177]: I'm always a bit worried about effects that cover many different reactive elements. And.. currently I'm wondering, what would happen if the update call was rejected with a 401? It feels like we could end up with lots of retries, and worst some loop.

Is there anything we could do to mitigate that?

**Review by mtsgrd — APPROVED** [review]

File: apps/desktop/src/lib/branch/SeriesHeader.svelte (hunk 1 above)
Comment (ndom91) [reply apps/desktop/src/lib/branch/SeriesHeader.svelte:177]: Yeah good call, I added a `catch` to the then chain and a boolean to stop looping

**Pull request merged**